deepchem/feat/base_classes.py

```diff
@@ -288,10 +297,15 @@ def featurize(self, datapoints, log_every_n=1000, **kwargs) -> np.ndarray:
         if isinstance(mol, str):
           # mol must be a RDKit Mol object, so parse a SMILES
           mol = Chem.MolFromSmiles(mol)
-          # SMILES is unique, so set a canonical order of atoms
-          new_order = rdmolfiles.CanonicalRankAtoms(mol)
-          mol = rdmolops.RenumberAtoms(mol, new_order)
-
+          try:
+            if not self.use_original_atom_ranks:  # condition if the original atom ranking in required
+              # SMILES is unique, so set a canonical order of atoms
+              new_order = rdmolfiles.CanonicalRankAtoms(mol)
+              mol = rdmolops.RenumberAtoms(mol, new_order)
+          except AttributeError:
+            # SMILES is unique, so set a canonical order of atoms
+            new_order = rdmolfiles.CanonicalRankAtoms(mol)
```

Comment:
Why are we just retrying the same operations in the exception handler?

Reply:
The original MolecularFeaturizer class doesn't have a `__init__()` function. I had to add it to create the instance variable of use_original_atom_ranks attached to self so that it could be used in the `featurize()`.

Since, MolecularFeaturizer class doesn't have a `__init__()` function, derived featurizer classes that have already been created with this class, do not call the this `__init__() ` in their `__init__()`. In those cases, the `self.use_original_atom_ranks` wouldn't have been declared. So that would have resulted in `AttributeError` if I had directly used the `self.use_original_atom_ranks` in `featurize()`.

To prevent this from happening, I used the exception handler. If ever the AttributeError occurs, that would mean that the derived class doesn't need `self.use_original_atom_ranks`. So, in exception part, I ran the original code, where the  datapoints are canonically reordered by default.